### PR TITLE
Adding in extension support

### PR DESCRIPTION
**Pull request by jwoertink**
This PR will be needed to do #122 and #188 

This adds in helper methods for enabling and disabling extensions.

```crystal
enable_extension "uuid-ossp"
disable_extension "uuid-ossp"
update_extension "hstore", to: "2.0"
```

### CODE REVIEW

File: spec/migrator/alter_extension_statement_spec.cr [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+describe Avram::Migrator::AlterExtensionStatement do
+  it "builds the proper SQL to update an extension" do
+    statement = Avram::Migrator::AlterExtensionStatement.new("uuid-ossp")
+    statement.build.should eq %{ALTER EXTENSION "uuid-ossp" UPDATE;}
+  end
+
+  it "updates to a specific version" do
+    statement = Avram::Migrator::AlterExtensionStatement.new("hstore", to: "2.0")
+    statement.build.should eq %{ALTER EXTENSION "hstore" UPDATE TO '2.0';}
```
Comment (paulcsmith) [inline spec/migrator/alter_extension_statement_spec.cr:11]: I didn’t know altering extensions was a thing! Nice that you added it

File: src/avram/migrator/alter_extension_statement.cr [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  def build
+    String.build do |sql|
+      sql << %{ALTER EXTENSION "#{@name}" UPDATE}
+      sql << to_version if @to
+      sql << ";"
+    end
+  end
+
+  def to_version
+    "TO '#{@to}'"
```
Comment (paulcsmith) [inline src/avram/migrator/alter_extension_statement.cr:14]:
```suggestion
    " TO '#{@to}'"
```

This should fix the spec failure

File: spec/migrator/alter_extension_statement_spec.cr (hunk 1 above)
Comment (jwoertink) [reply spec/migrator/alter_extension_statement_spec.cr:11]: There's actually a TON of altering you can do, but just specifying a specific version seemed to be the most common use I found. We can always extend this later if people start needing more options.

**Comment by jwoertink** [comment]
There's a few small options you could technically use like specifying the schema if it's not public, or installing an extension from a specific older version, but those seem to be super edge case since we currently don't support schemas outside of public, and I think you can just use the `update_extension` and specify a different version if you need to roll it back.

File: spec/migrator/alter_extension_statement_spec.cr (hunk 1 above)
Comment (paulcsmith) [reply spec/migrator/alter_extension_statement_spec.cr:11]: Sounds perfect to me! People can always drop down to `execute` if they need to 👍

Comment (jwoertink) [reply spec/migrator/alter_extension_statement_spec.cr:11]: Exactly. We have escape hatches for it, so this should cover the 90% 😄

**Pull request merged**